DSCResources/cChocoPackageInstall/cChocoPackageInstall.psm1

```diff
@@ -417,42 +417,54 @@ Function Upgrade-Package {
     $packageUpgradeOuput | ForEach-Object { Write-Verbose -Message $_ }
 
     # Clear Package Cache
-    Get-ChocoInstalledPackage 'Purge'
+    Get-ChocoInstalledPackage -Purge
 }
 
-function Get-ChocoInstalledPackage ($action) {
+function Get-ChocoInstalledPackage {
+    [CmdletBinding()]
+    param (
+        [switch]$Purge,
+        [switch]$NoCache
+    )
+
     $ChocoInstallLP = Join-Path -Path $env:ChocolateyInstall -ChildPath 'cache'
     if ( -not (Test-Path $ChocoInstallLP)){
         New-Item -Name 'cache' -Path $env:ChocolateyInstall -ItemType Directory | Out-Null
     }
     $ChocoInstallList = Join-Path -Path $ChocoInstallLP -ChildPath 'ChocoInstalled.xml'
 
-    if ($action -eq 'Purge') {
-        Remove-Item $ChocoInstallList -Force
-        $res = $true
-    } else {
+    if ( -not $Purge) {
         $PackageCacheSec = (Get-Date).AddSeconds('-60')
         if ( $PackageCacheSec -lt (Get-Item $ChocoInstallList -ErrorAction SilentlyContinue).LastWriteTime ) {
                 $res = Import-Clixml $ChocoInstallList
         } else {
-            choco list -lo -r | ConvertFrom-Csv -Header 'Name', 'Version' -Delimiter "|" -OutVariable res | Export-Clixml -Path $ChocoInstallList
+            $res = choco list -lo -r | ConvertFrom-Csv -Header 'Name', 'Version' -Delimiter "|"
+            if ( -not $NoCache){
+                $res | Export-Clixml -Path $ChocoInstallList
+            }
         }
+    } else {
+        Remove-Item $ChocoInstallList -Force
+        $res = $true
     }
 
     Return $res
 }
 
-function Get-ChocoVersion ($action) {
+function Get-ChocoVersion {
+    [CmdletBinding()]
+    param (
+        [switch]$Purge,
+        [switch]$NoCache
+    )
+
     $chocoInstallCache = Join-Path -Path $env:ChocolateyInstall -ChildPath 'cache'
     if ( -not (Test-Path $chocoInstallCache)){
         New-Item -Name 'cache' -Path $env:ChocolateyInstall -ItemType Directory | Out-Null
     }
     $chocoVersion = Join-Path -Path $chocoInstallCache -ChildPath 'ChocoVersion.xml'
 
-    if ($action -eq 'Purge') {
-        Remove-Item $chocoVersion -Force
-        $res = $true
-    } else {
+    if ( -not $Purge) {

```

Comment:
Same here?

Reply:
Thanks for reversing that - I think you missed removing the `-not` from this?

Reply:
gah.. I saw that right after I pushed the commit and fixed it but forgot to re-commit it.  proper fix coming in a second.

Reply:
Is testing $Purge.IsPresent, considered the correct way for test vs just `if ($Purge)`?  I was intending to update that but accidentally committed before being done making the requested updates.

Reply:
I think it's a personal taste because both work. For me, what I see `$Purge.IsPresent` I know `$Purge` is a switch. If I just see `$Purge` on it's own it could be anything.